### PR TITLE
fix(Checkbox): Allow children for Checkbox component

### PR DESCRIPTION
**Pull request by CPatchane**
Useful to pass much complicated text with bold parts or links

### CODE REVIEW

**Comment by CPatchane** [comment]
[(XS) [STORE] Pouvoir cliquer sur le texte à côté de la coche pour valider les CGUs](https://trello.com/c/aoLhuw7F/85-xs-store-pouvoir-cliquer-sur-le-texte-%C3%A0-c%C3%B4t%C3%A9-de-la-coche-pour-valider-les-cgus)

File: react/Checkbox/Readme.md [hunk 1]
```diff
@@ -17,3 +17,9 @@
 ```
 <div><Checkbox label="This is a mixed checkbox" mixed /></div>
 ```
+
+### Checkbox with complicated children
```
Comment (ptbrowne) [inline react/Checkbox/Readme.md]:
```suggestion
### Checkbox with complex children
```

https://english.stackexchange.com/questions/10459/what-is-the-difference-between-complicated-and-complex

File: react/Checkbox/Readme.md [hunk 2]
```diff
@@ -17,3 +17,9 @@
...
+
+```
+<div><Checkbox>This is a <strong>complicated</strong> text</Checkbox></div>
```
Comment (ptbrowne) [inline react/Checkbox/Readme.md]:
```suggestion
<div><Checkbox>This is a <strong>complex</strong> text</Checkbox></div>
```

File: react/__snapshots__/examples.spec.jsx.snap [hunk 3]
```diff
@@ -222,6 +222,12 @@ exports[`Checkbox should render examples: Checkbox 3`] = `
 </div>"
 `;
 
+exports[`Checkbox should render examples: Checkbox 4`] = `
+"<div>
+  <div><label class=\\"styles__c-input-checkbox___2CzvY\\" aria-checked=\\"\\"><input type=\\"checkbox\\" name=\\"\\" value=\\"\\"><span>This is a <strong>complicated</strong> text</span></label></div>
```
Comment (ptbrowne) [inline react/__snapshots__/examples.spec.jsx.snap]:
```suggestion
  <div><label class=\\"styles__c-input-checkbox___2CzvY\\" aria-checked=\\"\\"><input type=\\"checkbox\\" name=\\"\\" value=\\"\\"><span>This is a <strong>complex</strong> text</span></label></div>
```

**Pull request merged**